File: src/entities/objects/wall.ts
import { GameObject } from '@/entities/base';
import { COLOR_WALL } from '@/game/constants';

/**
 * Wall object - static solid objects
 */
export class Wall extends GameObject {
  constructor(x: number = 0, y: number = 0) {
    super(x, y);
    this.setProperty('solid', true);
    this.setProperty('pushable', false);
    this.setProperty('fragile', false);
    this.setProperty('supports_weight', true);
    this.setProperty('weight', Number.MAX_SAFE_INTEGER);
  }

  getType(): string {
    return 'wall';
  }

  getColor(): number {
    return COLOR_WALL;
  }

  update(_dt: number): void {
    // Walls don't need to update - they're static
  }
}

File: src/entities/objects/wall.py
# Wall Class

from src.entities.base import GameObject
from src.game.constants import COLOR_WALL


class Wall(GameObject):
    """Wall object that blocks movement"""
    
    def __init__(self, x: int = 0, y: int = 0):
        super().__init__(x, y)
        
        # Wall properties
        self.set_property('solid', True)
        self.set_property('pushable', False)
        self.set_property('fragile', False)
        self.set_property('affected_by_gravity', False)
        self.set_property('supports_weight', True)
        self.set_property('interactive', False)
        self.set_property('height', 1)  # Standard wall height
        self.set_property('weight', 999)  # Very heavy
    
    def get_type(self) -> str:
        return "wall"
    
    def get_color(self) -> tuple:
        return COLOR_WALL
    
    def can_be_jumped_over(self) -> bool:
        """Check if player can jump over this wall"""
        return self.get_property('height', 1) <= 1